### PR TITLE
Subscription Management: Create "Notify me of new comments" toggle

### LINKED ISSUE #76203: `/subscriptions/comments`: Create "Notify me of new comments" toggle
The goal of this task is to create a working "Notify me of new comments" toggle to the Comments page (**for logged-in users only**):

Image

Design: inDLaEQV8jJ21O4WXawIsJ-fi-712_26668

### PR DESCRIPTION
**Pull request by phcp**
Closes https://github.com/Automattic/wp-calypso/issues/76203

## Proposed Changes

* Create "Notify me of new comments" toggle
* Create `usePostNotifyMeOfNewCommentsMutation` hook
* Add tracks event for the toggle
* Fix action button position
* Change hover background color to match other menu items
* Fix fill color for icons on hover

image

## Testing Instructions

* Apply this PR to your local
* Go to http://calypso.localhost:3000/read/subscriptions/comments
* You should see the new toggle on the settings popover
* Toggle on/off and check if the value was updated correctly
* Toggle on and verify if the subscriber receives the comments notification

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/data-stores/src/reader/mutations/use-post-notify-me-of-new-comments-mutation.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+			const response = await callApi< PostSubscriptionNotifyMeOfNewCommentsResponse >( {
+				apiNamespace: 'wpcom/v2',
+				path: `/post-comment-subscriptions/${ params.subscriptionId }/update`,
+				method: 'POST',
+				body: {
+					send_comments: params.sendComments,
+				},
+				isLoggedIn,
+				apiVersion: '1.1',
```
Comment (JuanLucha) [inline packages/data-stores/src/reader/mutations/use-post-notify-me-of-new-comments-mutation.ts]: Mm... I think the version here should be 2.

**Review by JuanLucha — APPROVED** [review]
It works! I'll approve, but I'll give you my opinion on something we could improve: optimistic updates. As you can see in the video, the toggle goes through these states:
1. the user clicks the toggle
2. the update endpoint is called. The button is greyed as it's disable while this endpoint responds.
3. the button is not greyed anymore, as the update endpoint has returned the response already, but... it's value isn't updated yet.
4. the GET endpoint to retrieve the values of all the subscriptions is called and, when its response comes, then the value is updated.

https://github.com/Automattic/wp-calypso/assets/3832570/e2991c9e-21d7-42cc-add5-fe4dc3ac74bc

That moment between the button is enabled again but it's not updated yet, that's where optimistic updates come in handy. We can use the mutation to set the new value in the frontend before the endpoint gets, and then update it if necessary if the value returned is different for any reason.

File: packages/data-stores/src/reader/mutations/use-post-notify-me-of-new-comments-mutation.ts (hunk 1 above)
Comment (phcp) [reply packages/data-stores/src/reader/mutations/use-post-notify-me-of-new-comments-mutation.ts]: Good catch, I'll update it. Thanks!

**Comment by phcp** [comment]
> It works! I'll approve, but I'll give you my opinion on something we could improve: optimistic updates. As you can see in the video, the toggle goes through these states:

Interesting, I implemented the optimistic updates for this list, as you can see on the mutation hook. It should work (at least worked on previous tests), I'll check if I missed something. Thanks for the feedback @JuanLucha!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10053889

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @phcp for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10053889).

**Pull request merged**